spec/jobs/emit_web_hook_event_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe Jobs::EmitWebHookEvent do
+
+  let(:post_hook) { Fabricate(:web_hook) }
+  let(:inactive_hook) { Fabricate(:inactive_web_hook) }
+  let(:post) { Fabricate(:post) }
+  let(:user) { Fabricate(:user) }
+
+  it 'raises an error when there is no web hook record' do
+    expect { subject.execute(event_name: 'post') }.to raise_error(Discourse::InvalidParameters)
+  end
+
+  it 'raises an error when there is no event name' do
+    expect { subject.execute(web_hook_id: 1) }.to raise_error(Discourse::InvalidParameters)
+  end
+
+  it 'raises an error when event name is invalid' do
+    expect { subject.execute(web_hook_id: post_hook.id, event_name: 'post_random') }.to raise_error(Discourse::InvalidParameters)
+  end
+
+  it "doesn't emit when the hook is inactive" do
+    Jobs::EmitWebHookEvent.any_instance.expects(:web_hook_request).never
+    subject.execute(web_hook_id: inactive_hook.id, event_name: 'post', post_id: post.id)
+  end
+
+  it 'emits normally with sufficient arguments' do
+    Jobs::EmitWebHookEvent.any_instance.expects(:web_hook_request).once
+    subject.execute(web_hook_id: post_hook.id, event_name: 'post', post_id: post.id)
+  end
+
+  context 'with category filters' do
+    let!(:category) { Fabricate(:category) }
+    let!(:topic) { Fabricate(:topic) }
+    let!(:topic_with_category) { Fabricate(:topic, category_id: category.id) }
+    let!(:topic_hook) { Fabricate(:topic_web_hook, categories: [category]) }
```

Comment:
I personally dislike `let!` because it hides the side effects for future developers adding tests to this context. Instead of "give me everything", I prefer "let me choose what I need for this test".